StackExchange.Profiling/Timing.cs

```diff
@@ -114,13 +114,13 @@ public void RemoveMatchingChildren(Predicate<Timing> match)
         /// <see cref="CustomTiming"/> lists keyed by their type, e.g. "sql", "memcache", "redis", "http".
         /// </summary>
         [DataMember(Order = 6)]
-        public Dictionary<string, List<CustomTiming>> CustomTimings
+        public IReadOnlyDictionary<string, IReadOnlyList<CustomTiming>> CustomTimings
         {
             get
             {
                 lock(_lockObject)
                 {
-                    return _customTimings?.ToDictionary(p => p.Key, p => new List<CustomTiming>(p.Value));
+                    return _customTimings?.ToDictionary(p => p.Key, p => (IReadOnlyList<CustomTiming>)new List<CustomTiming>(p.Value));
```

Comment:
I think it would be better to get rid of all these copies on the get (here and Children, and ClientTimings)
I'd rather have explicity Set() methods and have the properties with no setters.